src/parsers/time-parser.js

```diff
@@ -31,8 +28,13 @@ export default class TimeParser {
       const start = date.start ? Number(date.start.date()) : null;
       const end = date.end ? Number(date.end.date()) : null;
       const extractedText = date.text;
-      const processedText = text.substr(0, date.index) +
-        text.substr(date.index + extractedText.length);
+      let beforeText = text.substr(0, date.index).trim();
+      const afterText = text.substr(date.index + extractedText.length).trim();
+      if (/ (?:on|by)$/.test(beforeText)) {
```

Comment:
Well, we're already in a custom chrono parser here. I thought it's the best place to do it because that's where we remove the text recognized by chrono.

maybe  we can file a bug on chrono to recognize "on/by" before a date as well, and add the bug number here ? What do you think ?